### PR TITLE
Add agile/1.0/board/{boardId}/issue

### PR DESCRIPTION
**Pull request by fuxx**
You can not receive all issues on a Kanban board with the rapidview REST API call. 
This only works for sprint boards. If you try to access a Kanban board through rapid view API you get an error which says that rapid board requires a project with a backlog.

To get all issues on a Kanban board the board API from agile module could be used. The /issue endpoint returns all known issues. I get all issues with a second call, equal to the implementation in rapidview.rb:+28.

This is my first pull request to a ruby project and my first project with rspec ever. I hope the stuff makes sense. Would be great to have a working gem soon :)

### CODE REVIEW

File: lib/jira/resource/agile.rb [hunk 1]
```diff
@@ -19,6 +19,18 @@ def self.get_backlog_issues(client, board_id, options = {})
         parse_json(response.body)
       end
 
+      def self.get_board_issues(client, board_id, options = {})
+        options[:maxResults] ||= 100
+        response = client.get(path_base(client) + "/board/#{board_id}/issue?#{hash_to_query_string(options)}")
+        parse_json(response.body)
```
Comment (SimonMiaou) [inline lib/jira/resource/agile.rb]: This line is useless as you do it the next line too.

File: spec/jira/resource/agile_spec.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
 require 'spec_helper'
 
 describe JIRA::Resource::Agile do
-  let(:client) { double(options: {rest_base_path: '/jira/rest/api/2', context_path: '/jira'}) }
+  let(:client) do
+    client = double(options: {rest_base_path: '/jira/rest/api/2', context_path: '/jira'})
+    allow(client).to receive(:Issue).and_return(JIRA::Resource::IssueFactory.new(client))
```
Comment (SimonMiaou) [inline spec/jira/resource/agile_spec.rb]: Why did you need this?

Comment (fuxx) [inline spec/jira/resource/agile_spec.rb]: If i dont akkiw the IssueFactory i cant access the IssueFactory and get a exception which says that :Issue is not available on client. I got this line from the filter_spec.rb:+6 and after insertion the :Issue worked on the client.

This is the exception i get
`RSpec::Mocks::MockExpectationError: #<Double (anonymous)> received unexpected message :Issue with (no args)
./lib/jira/resource/agile.rb:31:in `get_board_issues'
./spec/jira/resource/agile_spec.rb:37:in `block (3 levels) in <top (required)>'
-e:1:in `load'
-e:1:in `<main>'`

File: lib/jira/resource/agile.rb (hunk 1 above)
Comment (fuxx) [inline lib/jira/resource/agile.rb]: True! Thanks for the finding!

**Comment by fuxx** [comment]
I have updated the PR according to your comments :)

**Comment by fuxx** [comment]
@SimonMiaou Is the PR ok now or do I have todo anything else from your perspective?

**Comment by SimonMiaou** [comment]
Hey Fuxx

Sorry for the delay, I still want to solve the pagination issue that is not forwarded for the JQL but wanted to take a look before asking anything.

I'll take a moment at the end of the week as I don't have much time right now.

**Comment by SimonMiaou** [comment]
Hi fuxx,

I did a PR to you branch to solve the issue I was concerned about.
https://github.com/fuxx/jira-ruby/pull/1

Maybe we should also forward other parameters (ex: `expand`)
And therefor I'm wondering if this part should not be handle outside of the gem to avoid confusion about which parameter is forwarded and which is not.
So I would suggest to simply return the response from the API instead of return `Issue` objects

@bobbrodie What do you think?
